iam-token-manager/v1.ts

```diff
@@ -0,0 +1,198 @@
+   * is determined by the following logic:
+   * 1. If user provides their own managed access token, assume it is valid and send it
+   * 2. If this class is managing tokens and does not yet have one, make a request for one
+   * 3. If this class is managing tokens and the token has expired, refresh it
+   * 4. If this class is managing tokens and has a valid token stored, send it
+   *
+   * @param {Function} cb - callback function that the token will be passed to
+   */
+  public getToken(cb: Function) {
+    if (this.userAccessToken) {
+      // 1. use user-managed token
+      return cb(null, this.userAccessToken);
+    } else if (!this.tokenInfo.access_token) {
+      // 2. request an initial token
+      this.requestToken(null, (err, tokenResponse) => {
+        this.saveTokenInfo(tokenResponse);
+        return cb(err, this.tokenInfo.access_token);
+      });
+    } else if (this.isTokenExpired()) {
+      // 3. refresh a token
+      this.refreshToken(null, (err, tokenResponse) => {
+        this.saveTokenInfo(tokenResponse);
+        return cb(err, this.tokenInfo.access_token);
+      });
+    } else {
+      // 4. use valid managed token
+      return cb(null, this.tokenInfo.access_token);
+    }
+  }
+
+  /**
+   * Request an IAM token using an API key.
+   *
+   * @param {string} authHeader - The authorization header to send to the service. Defaults to 'Basic Yng6Yng='.
+   * @param {Function} cb - The callback that handles the response.
+   * @returns {void}
+   */
+  public requestToken(authHeader: string, cb: Function): void {
+    const authorization = authHeader || 'Basic Yng6Yng=';
+    const parameters = {
+      options: {
+        url: this.url,
+        method: 'POST',
+        headers: {
+          'Content-type': 'application/x-www-form-urlencoded',
+          Authorization: authorization
+        },
+        form: {
+          grant_type: 'urn:ibm:params:oauth:grant-type:apikey',
+          apikey: this.iam_apikey,
+          response_type: 'cloud_iam'
+        }
+      }
+    };
+    sendRequest(parameters, (error, body, response) => {
+      cb(error, body);
+    });
+  }
+
+  /**
+   * Refresh an IAM token using a refresh token.
+   *
+   * @param {string} authHeader - The authorization header to send to the service. Defaults to 'Basic Yng6Yng='.
+   * @param {Function} cb - The callback that handles the response.
+   * @returns {void}
+   */
+  public refreshToken(authHeader: string, cb: Function) {
+    const authorization = authHeader || 'Basic Yng6Yng=';
+    const parameters = {
+      options: {
+        url: this.url,
+        method: 'POST',
+        headers: {
+          'Content-type': 'application/x-www-form-urlencoded',
+          Authorization: authorization
+        },
+        form: {
+          grant_type: 'refresh_token',
+          refresh_token: this.tokenInfo.refresh_token
+        }
+      }
+    };
+    sendRequest(parameters, (error, body, response) => {
```

Comment:
write everything in one line.

```
sendRequest(parameters, cb);
```

It will send the response as the third parameter which is not bad actually.
We do it in the SDK already